### PR TITLE
BAU - Pull out common logic into trait

### PR DESCRIPTION
**Pull request by ed0906**
The Appeal, Review & Extended Use Status Change controllers have very similar logic. We could pull this out into a trait. Let me know your thoughts

### CODE REVIEW

**Comment by ed0906** [comment]
There's other options e.g.
- Instead of using inheritance, define a generic `updateStatus` method where we pass in various args required in the trait above
-  Leave it as it is

File: app/uk/gov/hmrc/tariffclassificationfrontend/controllers/StatusChangeAction.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        getCaseAndRenderView(
+          reference,
+          c => successful(chooseStatusView(c, errors))
+        )
+      },
+      (status: T) => {
+        getCaseAndRespond(
+          reference,
+          c => {
+            if(statusHasChanged(c, status)) {
```
Comment (pllaurado) [inline app/uk/gov/hmrc/tariffclassificationfrontend/controllers/StatusChangeAction.scala]: not about the refactor. But maybe move this logic into the service?

Comment (ed0906) [reply app/uk/gov/hmrc/tariffclassificationfrontend/controllers/StatusChangeAction.scala]: Potentially I guess the question is: Is checking if a status has changed business logic, or view layer?

I don't have a strong opinion either way

**Comment by gokyo** [comment]
I haven't looked at the code properly, but extracting the common code in a trait makes sense to me 👍

**Pull request merged**